doc/man3/OSSL_CRMF_MSG_get0_tmpl.pod

```diff
@@ -41,11 +44,17 @@ with the caller, who is responsible for freeing it.
 
 OSSL_CRMF_MSG_get_certReqId() retrieves the certReqId of B<crm>.
 
+OSSL_CRMF_ASN1_get_int() converts an ASN1_INTEGER into an int type.
+If successful (i.e., the ASN.1 encoded value is valid and fits in an int)
+it sets B<*pr> to the value of B<a>.
```

Comment:
I am not sure why this ASN.1 to int function is necessary at all.
At the very least it would need to be moved to `asn1.h`.
But note that `long ASN1_INTEGER_get(const ASN1_INTEGER *a)` was deprecated
because `long` is an implementation dependent type, just like `int`.
Is there a reason why `ASN1_INTEGER_get_int64` can't be used?

Reply:
Its virtue is the range check adapted to the size of int (rather than `int64`). As I wrote on Oct 18 (above): _"It's basically like `ASN1_INTEGER_get_int64()`, actually calling it, but with range checks according its result type int. It is used both in the CRMF and CMP sub-libraries. "_

I've meanwhile included is in the doc and added a TODO regarding move to asn1 (see the last commit). As I wrote also on Oct 18: _"BTW, this rather general function could also be moved (with adapted name) to `crypto/asn1/a_int.c`."_

Originally we used `long ASN1_INTEGER_get()` but stopped this because this function is deprecated. The reason for the deprecation is (according to a a statement in the man page, to which I agree) that it has a design flaw: if its result is -1 one cannot distinguish whether this is because of an input-out-of-range condition and the legitimate input value -1.

We could use `int64`, but 

1. `ASN1_INTEGER_get_int64()` is not available for OpenSSL v1.0.2 (for which unfortunately we need to provide compatibility code for legacy applications),
2. for application in CMP the guaranteed size of `int`, 32 bits, is sufficient, and
3. `int` saves space in our OSSL_CMP_CTX, where several integer values are contained.

If you still advise using `ASN1_INTEGER_get_int64()`, we can do (and use a different compatibility workaround for OpenSSL v1.0.2).

BTW, in any case we'll keep our contribution free of version switches, of course.

Reply:
Another issue with `long` in particular is that it depends on the platform.
And `int` is at least in theory also variable length, while `int32_t` is not.
But why is that function not an internal API, until it is decided whether there
will be an ASN1_get_int or ASN1_get_int32?

Reply:
And, by the way, where is the ASN1_INTEGER_set_int function?

Reply:
I like the idea of pushing `OSSL_CRMF_ASN1_get_int()` to the internal API, which would likely be the cleanest solution if we stick with this function.

We have meanwhile discussed the options and came to the conclusion that we simply move on to using instead the new function, `ASN1_INTEGER_get_int64()`. I'll update this PR accordingly.